src/zenml/integrations/aws/flavors/sagemaker_orchestrator_flavor.py

```diff
@@ -0,0 +1,110 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Amazon SageMaker orchestrator flavor."""
+
+from typing import TYPE_CHECKING, Dict, Optional, Type
+
+from zenml.config.base_settings import BaseSettings
+from zenml.integrations.aws import AWS_SAGEMAKER_STEP_OPERATOR_FLAVOR
+from zenml.orchestrators import BaseOrchestratorConfig
+from zenml.orchestrators.base_orchestrator import BaseOrchestratorFlavor
+
+if TYPE_CHECKING:
+    from zenml.integrations.aws.orchestrators import SagemakerOrchestrator
+
+
+class SagemakerOrchestratorSettings(BaseSettings):
+    """Settings for the Sagemaker orchestrator.
+
+    Attributes:
+        instance_type: The instance type to use for the processing job.
+        execution_role: The IAM role to use for the processing job.
+        volume_size_in_gb: The size of the EBS volume to use for the processing
+            job.
+        max_runtime_in_seconds: The maximum runtime in seconds for the
+            processing job.
+        processor_tags: Tags to apply to the processor assigned to the step.
+    """
+
+    instance_type: str = "ml.t3.medium"
+    execution_role: Optional[str] = None
+    volume_size_in_gb: Optional[int] = 30
+    max_runtime_in_seconds: Optional[int] = 86400
+    processor_tags: Optional[Dict[str, str]] = None
```

Comment:
I think all of these can be non-optional? For the tags you could simply make it have a default value of `{}`